### PR TITLE
SONARJAVA-1977: S2637 excludes default constructors for JPA entities.

### PR DESCRIPTION
**Pull request by Johnnei**
Please ensure your pull request adheres to the following guidelines: 

- [x] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [x] Unit tests are passing and you provided a unit test for your fix
- [x] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project.
- [x] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/se/checks/NonNullSetToNullCheck.java [hunk 1]
```diff
@@ -90,6 +96,18 @@ public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager co
...
   }
 
+  private static boolean isDefaultConstructorForJpa(MethodTree methodTree) {
+    if (!methodTree.block().body().isEmpty()) {
+      // Constructor does something.
+      return false;
+    }
+
+    ClassTree classTree = (ClassTree) methodTree.parent();
+    return classTree.modifiers().annotations().stream()
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/se/checks/NonNullSetToNullCheck.java]: It's probably easier to test presence of annotation by using symbol metadata:
```
SymbolMetadata metadata = classTree.symbol().metadata();
Stream.of(JPA_ANNOTATIONS).anyMatch(metadata::isAnnotatedWith);
```

Comment (Johnnei) [inline java-frontend/src/main/java/org/sonar/java/se/checks/NonNullSetToNullCheck.java]: Fixed it.

**Comment by Johnnei** [comment]
@m-g-sonar I've processed your review points. Time for review round two.

**Comment by m-g-sonar** [comment]
Thanks @Johnnei for the rework. Everything looks good to me now!
I'm going to open a PR duplicating your commits in order to trigger our internal QA. Once I'm sure everything is green, I'll merge it!

**Pull request merged**